frontend/packages/console-app/src/components/dashboards-page/status.ts

```diff
@@ -72,16 +72,16 @@ export const getControlPlaneComponentHealth = (
   return { state: HealthState.ERROR, message: perc.string };
 };
 
-const getWorstStatus = (
+export const getWorstStatus = (
   componentsHealth: SubsystemHealth[],
 ): { state: HealthState; message: string; count: number } => {
-  const withPriority = componentsHealth.map((h) => healthPriority[h.state]);
+  const withPriority = componentsHealth.map((h) => healthPriority?.[h.state] ?? {});
   const mostImportantState = Math.max(...withPriority.map(({ priority }) => priority));
```

Comment:
Maybe even better and less error prone would be to make sure that `healthPriority` contains all Health States. Currently its lacking Loading state. Once we add it I think there's no reason to have separate `healthStateMapping` and `healthPriority`. We just include the priority into `healthStateMapping` directly.

```typescript
export const healthStateMapping: { [key in HealthState]: PriorityHealthState } = {
  [HealthState.OK]: {
    priority: 0,
    health: HealthState.OK,
    icon: <GreenCheckCircleIcon />,
  },
  [HealthState.UNKNOWN]: {
    priority: 1,
    health: HealthState.UNKNOWN,
    icon: <GrayUnknownIcon />,
    message: 'Unknown',
  },
  [HealthState.PROGRESS]: {
    priority: 2,
    health: HealthState.PROGRESS,
    icon: <InProgressIcon />,
    message: 'Pending',
  },
  [HealthState.UPDATING]: {
    priority: 3,
    health: HealthState.UPDATING,
    icon: <BlueSyncIcon />,
    message: 'Updating',
  },
  [HealthState.WARNING]: {
    priority: 4,
    health: HealthState.WARNING,
    icon: <YellowExclamationTriangleIcon />,
    message: 'Degraded',
  },
  [HealthState.ERROR]: {
    priority: 5,
    health: HealthState.ERROR,
    icon: <RedExclamationCircleIcon />,
    message: 'Degraded',
  },
  [HealthState.LOADING]: {
    priority: 6,
    health: HealthState.LOADING,
    icon: <div className="skeleton-health" />,
  },
  [HealthState.NOT_AVAILABLE]: {
    priority: 7,
    health: HealthState.NOT_AVAILABLE,
    icon: <GrayUnknownIcon />,
    message: 'Not available',
  },
};
```

Can you try that ?